diff --git a/dist/node.mjs b/dist/node.mjs
index a12d50e19d1fa726246f9a2fbe6f76d07471c0ee..dcec4df6bba27e5c16df3034156a2cc263285228 100644
--- a/dist/node.mjs
+++ b/dist/node.mjs
@@ -49,2 +49,3 @@ const fetchClientMetadataResource = async (input, init) => {
-			lookup: (_hostname, _options, callback) => {
-				callback(null, pinnedAddress.address, pinnedAddress.family);
+			lookup: (_hostname, options, callback) => {
+				if (options.all) callback(null, [pinnedAddress]);
+				else callback(null, pinnedAddress.address, pinnedAddress.family);
